src/main/scala/rise/core/types/Flags.scala

```diff
@@ -0,0 +1,15 @@
+package rise.core.types
+
+object Flags {
+  /** This flag enables the type inference system to express explicit dependence
+    * of type variables within dependent functions.
+    *
+    * The pass is slow and interferes with advanced arithmetic simplification.
```

Comment:
it is not really a pass?

Reply:
Good point - will change to "transformation"